api/scaffold/__init__.py

```diff
@@ -1492,20 +1548,16 @@ class Chain(Module):
 
     __ADDR_CONTROL = 0x0900
 
-    def __init__(self, parent, index, size):
+    def __init__(self, parent: "Scaffold", index: int, size: int):
         """
-        :param parent: Scaffold instance owning the chain module.
-        :type parent: Scaffold
-        :param index: Module index.
-        :type index: int
+        :param parent: The :class:`Scaffold` instance owning the chain module.
+        :param index: Chain module index.
         :param size: Number of events in the chain.
-        :type size: int
         """
         super().__init__(parent, f"/chain{index}")
-        self.add_register("control", "wv", self.__ADDR_CONTROL + index * 0x10)
-        for i in range(size):
-            self.add_signal(f"event{i}")
-        self.add_signal("trigger")
+        self.reg_control = self.add_register("wv", self.__ADDR_CONTROL + index * 0x10)
+        self.events = self.add_signals(*[f"event{i}" for i in range(size)])
```

Comment:
Possible regression here, `scaffold.chain0.event0` is not possible anymore.
Although `scaffold.chain0.event[0]` is also good, existing scripts may break.
I suggest to keep this backward compatible.

Reply:
Adding this as a suggestion

```suggestion
        # For backward compatibility with scripts accessing events with scaffold.chainX.eventX
        for i, event in enumerate(self.events):
            self.__dict__[f"event{i}"] = event
```